databricks_cli/stack/api.py

```diff
@@ -91,6 +91,29 @@ def deploy(self, config_path, **kwargs):
         click.echo("Saving stack status to {}".format(status_path))
         self._save_json(status_path, new_stack_status)
 
+    def download(self, config_path, **kwargs):
+        """
+        Downloads a stack given stack JSON configuration template at path config_path.
+
+        Loads the JSON template as well as status JSON if stack has been deployed before.
+        Changes working directory to the same directory as where the config file is, then
+        calls on download_config to do the stack downloading.
+
+        The working directory is changed to that where the JSON template is contained
+        so that paths within the stack configuration are relative to the directory of the
+        JSON template instead of the directory where this function is called.
+
+        :param config_path: Path to stack JSON configuration template. Must have the fields of
+        'name', the name of the stack and 'resources', a list of stack resources.
+        :return: None.
+        """
+        stack_config = self._load_json(config_path)
+        config_dir = os.path.dirname(os.path.abspath(config_path))
+        cli_dir = os.getcwd()
+        os.chdir(config_dir)  # Switch current working directory to where json config is stored
+        self.download_config(stack_config, **kwargs)
```

Comment:
Why do you have to switch dirs in order to download? Is there a way to achieve this without switching dirs?

Reply:
This is the same process I'm using for deploying. I want the paths in the config file to be relative to where the config file is, so I switch to the directory of the config file. If the config file is in the current directory, then the directory isn't being switched.